toolchain/binutils/Makefile

```diff
@@ -43,8 +43,12 @@ HOST_CONFIGURE_ARGS = \
 	--host=$(GNU_HOST_NAME) \
 	--target=$(REAL_GNU_TARGET_NAME) \
 	--with-sysroot=$(TOOLCHAIN_DIR) \
+	--with-system-zlib \
```

Comment:
are we not able to specify a custom directory here like with gcc?

Reply:
That's just a bool flag AFAICS:
```
bfd/configure:with_system_zlib
bfd/configure:if test "${with_system_zlib+set}" = set; then :
bfd/configure:  withval=$with_system_zlib; if test x$with_system_zlib = xyes ; then
binutils/configure:with_system_zlib
binutils/configure:if test "${with_system_zlib+set}" = set; then :
binutils/configure:  withval=$with_system_zlib; if test x$with_system_zlib = xyes ; then
config/zlib.m4:  if test x$with_system_zlib = xyes ; then
configure:with_system_zlib
configure:if test "${with_system_zlib+set}" = set; then :
configure:  withval=$with_system_zlib;
configure:if test x$with_system_zlib = xyes ; then
configure:# Set with_gnu_as, with_gnu_ld, and with_system_zlib as appropriate.
configure:  with_system_zlib=yes
configure.ac:if test x$with_system_zlib = xyes ; then
configure.ac:# Set with_gnu_as, with_gnu_ld, and with_system_zlib as appropriate.
configure.ac:  with_system_zlib=yes
gas/configure:with_system_zlib
gas/configure:if test "${with_system_zlib+set}" = set; then :
gas/configure:  withval=$with_system_zlib; if test x$with_system_zlib = xyes ; then
gdb/configure:with_system_zlib
gdb/configure:if test "${with_system_zlib+set}" = set; then :
gdb/configure:  withval=$with_system_zlib; if test x$with_system_zlib = xyes ; then
gold/configure:with_system_zlib
gold/configure:if test "${with_system_zlib+set}" = set; then :
gold/configure:  withval=$with_system_zlib; if test x$with_system_zlib = xyes ; then
gprofng/configure:with_system_zlib
gprofng/configure:if test "${with_system_zlib+set}" = set; then :
gprofng/configure:  withval=$with_system_zlib; if test x$with_system_zlib = xyes ; then
ld/configure:with_system_zlib
ld/configure:if test "${with_system_zlib+set}" = set; then :
ld/configure:  withval=$with_system_zlib; if test x$with_system_zlib = xyes ; then
libctf/configure:with_system_zlib
libctf/configure:if test "${with_system_zlib+set}" = set; then :
libctf/configure:  withval=$with_system_zlib; if test x$with_system_zlib = xyes ; then
zlib/configure:with_system_zlib
zlib/configure:if test "${with_system_zlib+set}" = set; then :
zlib/configure:  withval=$with_system_zlib;
zlib/configure:   if test "$with_system_zlib" = yes; then
zlib/configure.ac:   if test "$with_system_zlib" = yes; then
```

But as I mentioned before, it gets HOST_*FLAGS to find our tools/ version.

zstd is found via pkg-config, which works because it includes toolchain-build.mk which includes host-build.mk which sets PKG_CONFIG_{PATH,LIB}:
```
bfd/configure:  --with-zstd             support zstd compressed debug sections
bfd/configure:# Check whether --with-zstd was given.
bfd/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
bfd/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
binutils/configure:  --with-zstd             support zstd compressed debug sections
binutils/configure:# Check whether --with-zstd was given.
binutils/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
binutils/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
config/zstd.m4:  [AS_HELP_STRING([--with-zstd], [support zstd compressed debug sections (default=auto)])],
config/zstd.m4:      AC_MSG_ERROR([--with-zstd was given, but pkgconfig/libzstd.pc is not found])
configure:  --with-zstd             Support zstd compressed debug sections
configure:# Check whether --with-zstd was given.
configure.ac:[AS_HELP_STRING([--with-zstd], [Support zstd compressed debug sections (default=auto)])])
gas/configure:  --with-zstd             support zstd compressed debug sections
gas/configure:# Check whether --with-zstd was given.
gas/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
gas/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
gdb/configure:  --with-zstd             support zstd compressed debug sections
gdb/configure:# Check whether --with-zstd was given.
gdb/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
gdb/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
gold/configure:  --with-zstd             support zstd compressed debug sections
gold/configure:# Check whether --with-zstd was given.
gold/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
gold/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
ld/configure:  --with-zstd             support zstd compressed debug sections
ld/configure:# Check whether --with-zstd was given.
ld/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
ld/configure:      as_fn_error $? "--with-zstd was given, but pkgconfig/libzstd.pc is not found" "$LINENO" 5
```

But since it's such a PITA to build gcc with zstd let's leave that for later...